docs/appendix/message-transport-mechanism.md

```diff
@@ -86,7 +90,22 @@ A rejection reason will be added to reject a message with an invalid version
 
 ### Service discovery
 
-In the service discovery stage, DNS[^16] is used to discover the capabilities as well as the endpoint host name and IP address of the remote participant.
+The purpose of the discovery stage is to discover the endpoint, public-key for signing and, optionally, the capabilities (e.g. supported UFTP versions) of the other participants.  
+The requirements for this stage rely heavily on agreements made between the participants that will be communicating with each other using the protocol.
+
+If service discovery fails due to inability to collect all required data within this timeout period, the delivery of the message fails.
+Such messages are removed from the outgoing queue and returned to the sending process, indicating a transport error.
+
+#### Single authority
+
+One possible implementation for service discovery is creating a single authority that administers all the relevant information, and provides this to participants.
+This approach could be seen as having a central address book, so all participants know where and how to send UFTP messages to each other. 
```

Comment:
I think it's good to use the new name (ShapeShifter) instead of UFTP

Reply:
updated